planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp

```diff
@@ -53,22 +54,21 @@ std::set<int64_t> getLaneIdSetOnPath(const autoware_planning_msgs::msg::PathWith
 
 }  // namespace
 
-IntersectionModuleManager::IntersectionModuleManager()
-: SceneModuleManagerInterface(getModuleName())
+IntersectionModuleManager::IntersectionModuleManager(rclcpp::Node & node)
+: SceneModuleManagerInterface(node, getModuleName())
 {
-  rclcpp::NodeHandle pnh("~");
   const std::string ns(getModuleName());
   auto & p = planner_param_;
-  pnh.param(ns + "/state_transit_mergin_time", p.state_transit_mergin_time, 2.0);
-  pnh.param(ns + "/decel_velocoity", p.decel_velocoity, 30.0 / 3.6);
-  pnh.param(ns + "/path_expand_width", p.path_expand_width, 2.0);
-  pnh.param(ns + "/stop_line_margin", p.stop_line_margin, 1.0);
-  pnh.param(ns + "/stuck_vehicle_detect_dist", p.stuck_vehicle_detect_dist, 5.0);
-  pnh.param(ns + "/stuck_vehicle_ignore_dist", p.stuck_vehicle_ignore_dist, 5.0) +
-    planner_data_->base_link2front;
-  pnh.param(ns + "/stuck_vehicle_vel_thr", p.stuck_vehicle_vel_thr, 3.0 / 3.6);
-  pnh.param(ns + "/intersection_velocity", p.intersection_velocity, 10.0 / 3.6);
-  pnh.param(ns + "/detection_area_length", p.detection_area_length, 200.0);
+  p.state_transit_mergin_time = node.declare_parameter(ns + "/state_transit_mergin_time", 2.0);
```

Comment:
It seems odd design that the manager declare parameters for the node. Shouldn't the node do that itself internally?

Reply:
I think it's fine – the individual modules know which parameters they require, so it makes sense to keep the parameter declaration close to where it's used.